types/types.go

```diff
@@ -240,3 +247,6 @@ type ClusterOrgKey struct {
 
 // NotifiedRecordsPerCluster maps a string representation of ClusterOrgKey to a NotificationRecord
 type NotifiedRecordsPerCluster map[ClusterOrgKey]NotificationRecord
+
+// NotifiedRecordsPerClusterByTarget let us split the notified records by their target (CCXDEV-8767)
+type NotifiedRecordsPerClusterByTarget map[EventTarget]NotifiedRecordsPerCluster
```

Comment:
I would add a different type, as one instance will only send notifications to one target, so it does not bring value, IMHO, to have a map of event targets

Reply:
then how can we deal with N targets? N types?

Reply:
My point is that assuming a job's instance only has one target, there should be no need to deal with N targets. Only the SELECT query's `event_type_id` should change. But that kind of assumptions is also the reason why we need to do so much refactoring now, so I won't even try to convince you if you don't agree with my reasoning =)

Reply:
IIRC @tisnik said that we should be able to:
- publish to Service Log and Notification Backend
- publish just to SL
- publish just to NB
- publish to none (for local testing f.e)
this is why I decided to use the map :grimacing:

Reply:
In that case, your approach is definitely good =)